R/drive_upload.R

```diff
@@ -66,35 +66,49 @@ drive_upload <- function(file = NULL,
   path <- path %||% root_folder()
   path <- as_dribble(path)
   confirm_single_file(path)
+  existing <- NULL
   if (!is_folder(path)) {
-    stop(
-      glue(
-        "Requested parent folder does not exist:\n{path$name}"
-      ),
-      call. = FALSE
-    )
+    if (!is.null(name)) {
+      stop(
+        glue(
+          "Requested parent folder does not exist:\n{path$name}"
+        ),
+        call. = FALSE
+      )
+    }
+    if (!overwrite) {
+      stop(
+        glue("File already exists:\n{path$name}"),
+        call. = FALSE
+      )
+    }
+    existing <- path
+    path <- unlist(promote(path, "parents")$parents)
```

Comment:
To be very clear, if we're updating media for existing id, we should NOT be touching parents. It would be easy to do more harm than good. In which case, I don't see why we need to store the parents.